### ISSUE TITLE
Log connection error in remote manager

### ISSUE THREAD

**Issue by osechet** [conan-io/conan#682]
For an unknown reason, I cannot connect the Conan server on my LAN from a Windows 7 machine. It was working before but now I get the following error when installing:
```
WARN: Unable to connect to myconan=http://myconan/
```
My others machine are still able to connect to the server.

Looking in the code, I found the message is printed when a `ConnectionError` is raised in `remote_manager.py` but there is no details about the error. 

It could be nice if the error was logged. At least I hope it could help me to understand why the connection is not possible anymore.

**Comment by memsharded** [comment]
Could you please post here the output of ``conan remote list``?
Is http://myconan/ mapped in your LAN? Just want to understand your setup. Have you tried pinging the machine from that Win 7 machine?
We can try to log the message too, but also to know what is happening and have it fixed without having to wait until next release.

**Comment by osechet** [comment]
The problem is fixed. As expected, it was a problem in the setup of the machine (since others machines were working fine). To be more precise, it was an issue with the proxy definition of the OS.
I still think to log the error would be a good thing since it could have help me to find the error faster.

**Comment by memsharded** [comment]
I have added and tested manually, and the result is something like:

```
WARN: Unable to connect to myconan=http://myconan/
HTTPConnectionPool(host='whatever', port=80): Max retries exceeded with url: /v1/conans/search
 (Caused by NewConnectionError('<requests.packages.urllib3.connection.HTTPConnection object at 
0x03E887D0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed',))
```

So, I'd say that the error helps very little, and it adds noise and confusion about the possible source of error. I think it is better to leave it as is, what do you think?

**Comment by osechet** [comment]
I'm now able to reproduce the error on any OS with any conan server. 

- On Windows, in the control panel, set a proxy to a fake ip. 
- On Linux, export the `http_proxy` and `https_proxy` variables (i.e. `export http_proxy=http://not.a.proxy:8080; export https_proxy=http://not.a.proxy:8080`)

Then call `conan search -r conan.io`. Since conan cannot go through the given proxy, it raises a `ConnectionError`. I tried from the sources and logged the error and I get:
`ERROR :remote_manager.py[154]: HTTPSConnectionPool(host='server.conan.io', port=443): Max retries exceeded with url: /v1/conans/search (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7fdc96cb2610>: Failed to establish a new connection: [Errno -2] Name or service not known',))) [2016-11-23 11:07:31,995]`

It's quite verbose but I can see that the error is caused by a `ProxyError`. With such a log, I would have found my mistake faster.

**Comment by memsharded** [comment]
Released in 0.16.
